nova_vm/src/ecmascript/builtins/indexed_collections/typed_array_objects/typed_array_intrinsic_object.rs

```diff
@@ -3788,3 +3946,104 @@ fn sort_comparator_typed_array<'a, T: Viewable + Copy + std::fmt::Debug>(
     slice[..copy_len].copy_from_slice(&items[..copy_len]);
     Ok(())
 }
+
+fn filter_typed_array<'a, T: Viewable + 'static + std::fmt::Debug>(
+    agent: &mut Agent,
+    callback: Function<'_>,
+    this_arg: Value,
+    o: TypedArray,
+    len: i64,
+    mut gc: GcScope<'a, '_>,
+) -> JsResult<'a, TypedArray<'a>> {
+    let callback = callback.bind(gc.nogc());
+    let this_arg = this_arg.bind(gc.nogc());
+    let o = o.bind(gc.nogc());
+    let callback = callback.scope(agent, gc.nogc());
+    let this_arg = this_arg.scope(agent, gc.nogc());
+    let scoped_o = o.scope(agent, gc.nogc());
+    // 5. Let kept be a new empty List.
+    // 6. Let captured be 0.
+    let mut kept: Vec<T> = Vec::with_capacity(len.try_into().unwrap());
+    // 7. Let k be 0.
+    // 8. Repeat, while k < len,
+    // b. Let kValue be ! Get(O, Pk).
+    let ta = scoped_o.get(agent);
+    let byte_offset = ta.byte_offset(agent);
+    let byte_length = ta.byte_length(agent);
+    let array_buffer = ta
```

Comment:
Let's do this:
```suggestion
    let local_array_buffer = ta.get_viewed_array_buffer(agent, gc.nogc());
    let array_buffer = local_array_buffer.scope(agent, gc.nogc());
    let properly_aligned = unsafe { local_array_buffer.as_slice(agent).align_to::<T>().0.is_empty() };
    if !properly_aligned {
      return Err(agent.throw_exception_with_static_message(
        ExceptionType::TypeError,
        "TypedArray is not properly aligned",
        gc.into_nogc(),
      ));
    }
```

With this we can skip checking the alignment on every loop iteration.